pkg/collector/corechecks/snmp/internal/checkconfig/profile.go

```diff
@@ -120,6 +120,13 @@ func loadProfiles(pConfig profileConfigMap) (profileConfigMap, error) {
 			}
 			profConfig.Definition = *profDefinition
 		}
+		profiledefinition.NormalizeMetrics(profConfig.Definition.Metrics)
+		errors := validateEnrichMetadata(profConfig.Definition.Metadata)
+		errors = append(errors, ValidateEnrichMetrics(profConfig.Definition.Metrics)...)
+		errors = append(errors, ValidateEnrichMetricTags(profConfig.Definition.MetricTags)...)
+		if len(errors) > 0 {
+			return nil, fmt.Errorf("validation errors: %s", strings.Join(errors, "\n"))
+		}
```

Comment:
Right now, we don't apply the normalize/validate/enrich when `profConfig.DefinitionFile` is nil (and profConfig.Definition is already populated).

This change will help handle this case by moving those processing closer to `profiles[name] = profConfig`